src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt

```diff
@@ -35,32 +35,76 @@ class OpenFileLoadHandlerGenerator(snykCodeFile: SnykCodeFile) {
+                        document.documentElement.style.setProperty('--disabled-text-color', style.getPropertyValue('--disabled-text-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--selected-text-color', style.getPropertyValue('--selected-text-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--error-text-color', style.getPropertyValue('--error-text-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--data-flow-file-color', style.getPropertyValue('--data-flow-file-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--data-flow-body-color', style.getPropertyValue('--data-flow-body-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--example-line-added-color', style.getPropertyValue('--example-line-added-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--example-line-removed-color', style.getPropertyValue('--example-line-removed-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--tabs-bottom-color', style.getPropertyValue('--tabs-bottom-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--tab-item-color', style.getPropertyValue('--tab-item-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--tab-item-hover-color', style.getPropertyValue('--tab-item-hover-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--tab-item-icon-color', style.getPropertyValue('--tab-item-icon-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--scrollbar-thumb-color', style.getPropertyValue('--scrollbar-thumb-color-${if (isDarkTheme) "dark" else "light"}'));
+                        document.documentElement.style.setProperty('--scrollbar-color', style.getPropertyValue('--scrollbar-color-${if (isDarkTheme) "dark" else "light"}'));
+
+                        if (${isDarculaTheme}) {
+                            document.documentElement.style.setProperty('--data-flow-body-color', style.getPropertyValue('--data-flow-body-color-darcula'));
+                            document.documentElement.style.setProperty('--example-line-added-color', style.getPropertyValue('--example-line-added-color-darcula'));
+                            document.documentElement.style.setProperty('--example-line-removed-color', style.getPropertyValue('--example-line-removed-color-darcula'));
+                        }
+                        })();
```

Comment:
I'd like to find a better way to handle this maybe by creating a mapping of theme-related properties that can be iterated over 🤔

Reply:
Also, I'm still investigating how to move this logic to the Language Server template where we'd be identifying the IDE via a variable sent by the IDE. For IntelliJ, it would be something like:

```kotlin
browser.executeJavaScript("window.hostIDE = 'jetbrains';", browser.url, 0)
```

So we will be handling theme specific for each IDE since VSCode needs custom properties like `--vscode-editor-foreground`

Still WIP 🏗️

Reply:
If it's IDE specific - does it need to be moved to LS? If it's easier to code & maintain, sure. But IDE-specific stuff could easily stay in the IDE.

Reply:
Regarding the `if (isDarktheme)...` I'd just define a variable `theme` that you can add, so you don't clutter the styles with inline `if` clauses.

Reply:
Hm... for custom style properties (e.g. --vscode-editor-foreground), could they just be template variables to be substituted in the IDE?